src/frontend/src/components/SearchBar/SearchBar.jsx

```diff
@@ -96,11 +96,13 @@ function CustomizedInputBase(props) {
 
   const onTextChange = (event) => {
     onChangeHandler(event.target.value);
+    navigate(`/search?q=${event.target.value}`);
   };
 
   const onSubmit = (event) => {
     event.preventDefault();
     onFormSubmit();
+    navigate(`/search?q=${searchText}&f=${filter}`);
```

Comment:
Why deviate from `text` and `filter`?  This seems like it will cause confusion between frontend and backend.

Reply:
Cause "/search?q={...}" is the standard query string, I thought it could be easily to be understood by others. I could make it consistence with our backend.